File: bindings/typescript/examples/invoke_wait.ts
/**
 * invoke + wait — the "runtime as a function" path.
 *
 * Run a published recipe and block for its committed result. Start a gateway
 * first: `kx serve --dev-allow-local --journal /tmp/kx.db --content /tmp/kx-blobs`.
 *
 *   npx tsx examples/invoke_wait.ts
 */

import { KxClient, type Result } from "@kortecx/sdk";

async function main(): Promise<void> {
  const kx = new KxClient("http://127.0.0.1:50151");
  try {
    const result = (await kx.invoke(
      "kx/recipes/echo",
      { topic: "hello" },
      { wait: true },
    )) as Result;
    console.log("state           :", result.state);
    console.log("instance_id     :", result.instanceId);
    console.log("terminal_mote_id:", result.terminalMoteId);
    console.log("result          :", result.text);
  } finally {
    kx.close();
  }
}

main().catch((err) => {
  console.error(String(err));
  process.exitCode = 1;
});

File: bindings/python/examples/invoke_wait.py
"""Call the runtime like a function: invoke a recipe and wait for the result.

Start a gateway first:
    kx serve --journal /tmp/kx.db --content /tmp/kx-blobs --dev-allow-local

Then:
    python examples/invoke_wait.py
"""

from __future__ import annotations

from kortecx import KxClient


def main() -> None:
    with KxClient("http://127.0.0.1:50151") as kx:
        result = kx.invoke("kx/recipes/echo", {"topic": "hello, kortecx"}, wait=True)
        print("state           ", result.state)
        print("instance_id     ", result.instance_id)
        print("terminal_mote_id", result.terminal_mote_id)
        print("result_ref      ", result.result_ref)
        # The demo recipe's bytes are binary; .text is None, .bytes has the payload.
        print("result bytes    ", (result.bytes or b"")[:32], "…")


if __name__ == "__main__":
    main()
